service/frontend/workflow_handler.go

```diff
@@ -4749,6 +4750,358 @@ func (wh *WorkflowHandler) StartBatchOperation(
+		operation.UpdateActivityOptionsOperation.Identity = op.UpdateActivityOptionsOperation.GetIdentity()
+		if updateMask := op.UpdateActivityOptionsOperation.GetUpdateMask(); updateMask != nil {
+			operation.UpdateActivityOptionsOperation.UpdateMask = updateMask
+		}
+		if ao := op.UpdateActivityOptionsOperation.GetActivityOptions(); ao != nil {
+			operation.UpdateActivityOptionsOperation.ActivityOptions = &activitypb.ActivityOptions{
+				ScheduleToStartTimeout: ao.ScheduleToStartTimeout,
+				ScheduleToCloseTimeout: ao.ScheduleToCloseTimeout,
+				StartToCloseTimeout:    ao.StartToCloseTimeout,
+				HeartbeatTimeout:       ao.HeartbeatTimeout,
+			}
+			if rp := ao.RetryPolicy; rp != nil {
+				operation.UpdateActivityOptionsOperation.ActivityOptions.RetryPolicy = &commonpb.RetryPolicy{
+					InitialInterval:        rp.InitialInterval,
+					MaximumInterval:        rp.MaximumInterval,
+					BackoffCoefficient:     rp.BackoffCoefficient,
+					NonRetryableErrorTypes: rp.NonRetryableErrorTypes,
+					MaximumAttempts:        rp.MaximumAttempts,
+				}
+			}
+			operation.UpdateActivityOptionsOperation.ActivityOptions.TaskQueue = ao.TaskQueue
+		}
+		batchOperation.Operation = operation
+	default:
+		return nil, serviceerror.NewInvalidArgumentf("The operation type %T is not supported", op)
+	}
+
+	// inputPayload, err := payloads.Encode(batchOperation)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	memo := &commonpb.Memo{
+		Fields: map[string]*commonpb.Payload{
+			batcher.BatchOperationTypeMemo: payload.EncodeString(operationType),
+			batcher.BatchReasonMemo:        payload.EncodeString(request.GetReason()),
+		},
+	}
+
+	// Add pre-define search attributes
+	var searchAttributes *commonpb.SearchAttributes
+	searchattribute.AddSearchAttribute(&searchAttributes, searchattribute.BatcherUser, payload.EncodeString(identity))
+	searchattribute.AddSearchAttribute(&searchAttributes, searchattribute.TemporalNamespaceDivision, payload.EncodeString(batcher.NamespaceDivision))
+
+	startReq := &workflowservice.StartBatchWorkflowExecutionRequest{
+		Namespace:                request.Namespace,
+		WorkflowId:               request.GetJobId(),
+		WorkflowType:             &commonpb.WorkflowType{Name: batcher.BatchWFTypeName},
+		TaskQueue:                &taskqueuepb.TaskQueue{Name: primitives.PerNSWorkerTaskQueue},
+		Input:                    batchOperation,
+		Identity:                 identity,
+		RequestId:                uuid.New(),
+		WorkflowIdConflictPolicy: enumspb.WORKFLOW_ID_CONFLICT_POLICY_FAIL,
+		WorkflowIdReusePolicy:    enumspb.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
+		Memo:                     memo,
+		SearchAttributes:         searchAttributes,
+		Priority:                 &commonpb.Priority{}, // ie default priority
+	}
+
+	// NOTE seankane: this is where the issue is, cannot figure out how to create a new method on historyclient
```

Comment:
How do I add a method onto the history client for the new batch operation? @bergundy @cretz

Reply:
@spkane31 let's sync offline.